src/SuperVault.sol

```diff
@@ -116,46 +122,67 @@ contract SuperVault is BaseStrategy, ISuperVault {
     }
 
     // @inheritdoc ISuperVault
-    function rebalance(
-        uint256[] memory superformIdsRebalanceFrom,
-        uint256[] memory amountsRebalanceFrom,
-        uint256[] memory superformIdsRebalanceTo,
-        uint256[] memory weightsOfRedestribution,
-        uint256 rebalanceFromMsgValue,
-        uint256 rebalanceToMsgValue,
-        uint256 slippage
-    )
-        external
-        payable
-        override
-        onlySuperVaultsStrategist
-    {
+    function rebalance(RebalanceArgs memory a) external payable override onlySuperVaultsStrategist {
+        uint256 lenRebalanceFrom = a.superformIdsRebalanceFrom.length;
+        uint256 lenFinal = a.finalSuperformIds.length;
         // Validate input arrays
-        if (
-            superformIdsRebalanceFrom.length != amountsRebalanceFrom.length
-                || superformIdsRebalanceTo.length != weightsOfRedestribution.length
-        ) {
+        if (lenRebalanceFrom != a.amountsRebalanceFrom.length || lenFinal != a.weightsOfRedestribution.length) {
             revert ARRAY_LENGTH_MISMATCH();
         }
+        uint256 numberOfSuperforms = SV.numberOfSuperforms;
 
-        // Check if any ID in rebalanceFrom is present in rebalanceTo
-        for (uint256 i = 0; i < superformIdsRebalanceFrom.length; i++) {
-            for (uint256 j = 0; j < superformIdsRebalanceTo.length; j++) {
-                if (superformIdsRebalanceFrom[i] == superformIdsRebalanceTo[j]) {
-                    revert DUPLICATE_SUPERFORM_ID();
+        // Check if superformIdsRebalanceFrom existin SVData and save the indexes of the found ids
+        uint256 foundCount = 0;
+        bool[] memory foundInSV = new bool[](lenRebalanceFrom);
+        for (uint256 i; i < lenRebalanceFrom; ++i) {
```

Comment:
Need duplicate check in here?

Reply:
maybe instead of passing in the rebalanceFrom ids and checking it if it exists in the SV storage data, we could just accept rebalanceAmounts and accept `0` for superformIds that don't need to be rebalanced